an-rope/src/lib.rs

```diff
@@ -1227,9 +1227,7 @@ impl iter::Extend<char> for Rope{
 
     fn extend<T>(&mut self, iter: T)
     where T: IntoIterator<Item=char> {
-        let s: String = iter.into_iter().collect();
-        let r: Rope = Rope::from(s);
-        self.append(r);
+        self.append(iter.into_iter().collect::<Rope>());
```

Comment:
You actually don't need the type annotation here. Since `append()` _only_ takes a `Rope`, the compiler is smart enough to figure out that `collect()` should return a `Rope`. So you can just say
```rust
self.append(iter.into_iter().collect());
```